src/SharedEventManager.php

```diff
@@ -177,23 +177,39 @@ public function getListeners(array $identifiers, $eventName)
                 ));
             }
 
-            $listenersByIdentifier = isset($this->identifiers[$identifier]) ? $this->identifiers[$identifier] : [];
-
-            $listeners = array_merge_recursive(
-                $listeners,
-                isset($listenersByIdentifier[$eventName]) ? $listenersByIdentifier[$eventName] : [],
-                isset($listenersByIdentifier['*']) ? $listenersByIdentifier['*'] : []
-            );
+            if (isset($this->identifiers[$identifier])) {
+                $listenersByIdentifier = $this->identifiers[$identifier];
+                if (isset($listenersByIdentifier[$eventName])) {
+                    foreach ($listenersByIdentifier[$eventName] as $p => $l) {
+                        $listeners[$p][] = $l;
+                    }
+                }
+                if (isset($listenersByIdentifier['*'])) {
+                    foreach ($listenersByIdentifier['*'] as $p => $l) {
+                        $listeners[$p][] = $l;
+                    }
+                }
+            }
         }
 
-        if (isset($this->identifiers['*']) && ! in_array('*', $identifiers)) {
+        if (isset($this->identifiers['*'])) {
             $wildcardIdentifier = $this->identifiers['*'];
+            if (isset($wildcardIdentifier[$eventName])) {
+                foreach ($wildcardIdentifier[$eventName] as $p => $l) {
+                    $listeners[$p][] = $l;
+                }
+            }
+            if (isset($wildcardIdentifier['*'])) {
+                foreach ($wildcardIdentifier['*'] as $p => $l) {
+                    $listeners[$p][] = $l;
+                }
+            }
+        }
 
-            $listeners = array_merge_recursive(
-                $listeners,
-                isset($wildcardIdentifier[$eventName]) ? $wildcardIdentifier[$eventName] : [],
-                isset($wildcardIdentifier['*']) ? $wildcardIdentifier['*'] : []
-            );
+        foreach ($listeners as $p => $l) {
+            // Argument unpacking requires PHP-5.6
```

Comment:
Oh, wait... package still supports PHP 5.5... @weierophinney are you fine with dropping 5.5 for 3.1.0?